app/models/ability.rb

```diff
@@ -0,0 +1,65 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    can :manage, :all if user.is_admin?
+    cannot :impersonate, User unless user.is_webauth_admin?
+
+    can [:manage_item, :manage_content, :manage_desc_metadata], ActiveFedora::Base if user.is_manager?
+    can :create, Dor::AdminPolicyObject if user.is_manager?
+
+    can [:view_metadata, :view_content], ActiveFedora::Base if user.is_viewer?
+
+    can :manage_item, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_item? user.roles(dor_item.pid)
+    end
+
+    can :manage_content, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_content? user.roles(dor_item.pid)
+    end
+
+    can :manage_desc_metadata, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_manage_desc_metadata? user.roles(dor_item.pid)
+    end
+
+    can :manage_item, ActiveFedora::Base do |dor_item|
+      if dor_item.admin_policy_object
+        dor_item.can_manage_item? user.roles(dor_item.admin_policy_object.pid)
+      end
+    end
+
+    can :manage_content, ActiveFedora::Base do |dor_item|
+      if dor_item.admin_policy_object
+        dor_item.can_manage_content? user.roles(dor_item.admin_policy_object.pid)
+      end
+    end
+
+    can :manage_desc_metadata, ActiveFedora::Base do |dor_item|
+      if dor_item.admin_policy_object
+        dor_item.can_manage_desc_metadata? user.roles(dor_item.admin_policy_object.pid)
+      end
+    end
+
+    can :view_content, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_view_content? user.roles(dor_item.pid)
+    end
+
+    can :view_metadata, Dor::AdminPolicyObject do |dor_item|
+      dor_item.can_view_metadata? user.roles(dor_item.pid)
+    end
```

Comment:
For 45-51, it's not clear that an APO with a governing APO will first query for permission from the governing APO.  I would @LynnMcRae to clarify the governance logic, but I believe it should be similar to 
https://github.com/sul-dlss/argo/blob/user-permissions/app/models/user.rb#L219-L228

To summarize, any DOR object with a "governing-APO" must always query the "governing-APO" for a permission.  Only when that fails and the object is an APO itself, should it be able to query itself directly for permission.  If so, it looks like these blocks should include something like:
```
if dor_item.admin_policy_object
  dor_item.can_manage_something? user.roles(dor_item.admin_policy_object.pid)
end
```